airflow/providers/amazon/aws/hooks/s3.py

```diff
@@ -57,8 +57,14 @@ def wrapper(*args, **kwargs) -> T:
 
         if "bucket_name" not in bound_args.arguments:
             self = args[0]
-            if self.conn_config and self.conn_config.schema:
-                bound_args.arguments["bucket_name"] = self.conn_config.schema
```

Comment:
We need also provide backward compatibility with current version.
Basically we need to check `bucket_name` in both places:
1. In `service_config.s3.bucket_name`
2. Fallback to `conn_config. schema` and depreciation warning

This also could be done in  [AwsConnectionWrapper](https://github.com/apache/airflow/blob/8290ade26deba02ca6cf3d8254981b31cf89ee5b/airflow/providers/amazon/aws/utils/connection_wrapper.py#L76)

Reply:
That makes sense. I have committed changes to include the backwards functionality. Should I also include a test to make sure the backward functionality still works, in `test_s3.py`?